### ISSUE TITLE
Brain File confusion

### ISSUE THREAD

**Issue by JamesCann** [Unity-Technologies/ml-agents#3213]
So, I've been able to look at a few examples. I have the latest as of this email.

This may sound like a twilight zone scenario as something doesn't add up.

I was training my model and it was all going fine. All of a sudden, I started getting an error (see below)

You see, I never had a Model selected in the inspector at first and it was training fine. 
I copied the nn file to the project and selected it. I tried training with the nn was selected, no luck in the command window, it doesn't look like it's training and it times out.
So, I remove the nn file and try again.
Now for some reason, with the model "blank" it won't let me train (I did before)

How do you train a model first without having one to begin with? (frustrated tonight)

NullReferenceException: Object reference not set to an instance of an object
MLAgents.InferenceBrain.ModelRunner.DecideBatch () (at Assets/ML-Agents/Scripts/InferenceBrain/ModelRunner.cs:129)
MLAgents.BarracudaPolicy.DecideAction () (at Assets/ML-Agents/Scripts/Policy/BarracudaPolicy.cs:53)
MLAgents.Agent.DecideAction () (at Assets/ML-Agents/Scripts/Agent.cs:1014)
MLAgents.Academy.EnvironmentStep () (at Assets/ML-Agents/Scripts/Academy.cs:323)
MLAgents.Academy.FixedUpdate () (at Assets/ML-Agents/Scripts/Academy.cs:350)

**Comment by Roskavaki** [comment]
Is the Agent set to the "default" option (instead of "inference" or "heuristic only") ?

**Comment by JamesCann** [comment]
no, its set to inference

**Comment by JamesCann** [comment]
I used to be able to connect to the trainer but no longer works. It's not like I changed anything, it just stopped working.

I get this...

INFO:mlagents_envs:Listening on port 5004. Start training by pressing the Play button in the Unity Editor.
INFO:mlagents.trainers:UnityEnvironment worker 0: environment stopping.

If I dont run the trainer Unity complains it can't connect to 5004 as an info line on the Unity Console.
